assignments/closure.js

```diff
@@ -1,18 +1,44 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
 
+function dragon(){
+  const egg=7;
+  function chicken(){
+    console.log(`I just ate the dragons' ${egg} eggs`);
+  }
+  chicken();
+}
+dragon();
+
+
+
 
 /* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
 
 
 // ==== Challenge 2: Create a counter function ====
-const counter = () => {
-  // Return a function that when invoked increments and returns a counter variable.
-};
+// const counter = () => {
+//   // Return a function that when invoked increments and returns a counter variable.
+// };
 // Example usage: const newCounter = counter();
 // newCounter(); // 1
 // newCounter(); // 2
 
+function counter(){
+  let counter=0;
+  return function currentCount(){
+    counter ++;
+    return counter;
+  }
```

Comment:
To emphasize that  `++` is a unary operator (only needs one operand - works with just one value), don't leave whitespace between it and the word `counter` (for readability).